lib/matplotlib/axes/_axes.py

```diff
@@ -2350,18 +2350,20 @@ def stem(self, *args, linefmt=None, markerfmt=None, basefmt=None,
             which inspired this method.
 
         """
+        if len(args) > 5:
+            raise TypeError('Too many positional arguments')
+
         # Assume there's at least one data array
         y = np.asarray(args[0])
         args = args[1:]
 
         # Try a second one
         try:
-            x, y = y, np.asarray(args[0], dtype=float)
-        except (IndexError, ValueError):
-            # The second array doesn't make sense, or it doesn't exist
+            x, *args = args
+        except IndexError:
```

Comment:
There isn't any indexing anymore, or is this what python raises when it can't unpack?